### PR TITLE
add user setting to select a default label printer (#4349)

### LINKED ISSUE #4349: [FR] Setting to define a default label printer
### Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find a similar feature request

### Problem statement

It would be very convenient to have an option to define a default label printer in the settings. I am using the `inventree-brother-plugin` heavily for all label making where it is a minor hassle to always have to manually change the selection from `Export as PDF` to `Brother Labels - Brother Label Printer` in the label printing modal form.

### Suggested solution

Some kind of configuration setting would be nice - could be a simple "make this printer the default selection" toggle somewhere.
It is unclear to me where to put this, as I am unsure where and how the brother plugin registers itself. Would the main `Label Printing` server configuration page be aware of the available printers? A drop-down to select the default printer seems logical to me.

Or is it a setting that the plugin should enforce somehow? (But that would potentially interfere with multiple plugins that each provide printers).

I am happy to give it a go myself with some pointers regarding these questions...

### Describe alternatives you've considered

This could be a user-specific setting as well - but I think since this is more of an "infrastructure/hardware" setup and should be in the server settings...

### Examples of other systems

_No response_

### Do you want to develop this?

- [X] I want to develop this.

### PR DESCRIPTION
**Pull request by simonkuehling**
This is my initial implementation for the "default label printer" user setting - it is working like intended so far already, but I'm open for comments...

Closes #4349

### CODE REVIEW

**Review by matmair — APPROVED** [review]

**Comment by simonkuehling** [comment]
I have run a bit of "monkey testing" the edge cases like deactivating a plugin after it was configured as default printer and such, and everything is behaving correctly - so I would see this as finished from my side...

File: InvenTree/common/models.py [hunk 1]
```diff
@@ -1628,6 +1629,15 @@ def requires_restart(self):
             return False
 
 
+def label_printer_options():
+    """Build a list of available label printer options."""
+    printers = [('', _('no printer (Export to PDF)'))]
```
Comment (SchrodingersGat) [inline InvenTree/common/models.py:1634]: Small thing, please capitalize "No Printer"

**Review by SchrodingersGat — CHANGES_REQUESTED** [review]

**Comment by SchrodingersGat** [comment]
@simonkuehling this works really well, just a small change to the text :)

File: InvenTree/common/models.py (hunk 1 above)
Comment (simonkuehling) [reply InvenTree/common/models.py]: good catch - that would be terribly inconsistent :facepalm: - fixed it!

**Comment by wolflu05** [comment]
Btw. you don't need to force push, as they anyway merge via squash so history will be clean in the end.

**Review by SchrodingersGat — APPROVED** [review]

**Review by matmair — APPROVED** [review]
Lgtm

**Comment by SchrodingersGat** [comment]
Thanks again @simonkuehling

**Pull request merged**